sdk/translation/ai-translation-text-rest/src/authentication.ts

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { AzureKeyCredential } from "@azure/core-auth";
+import {
+  PipelinePolicy,
+  PipelineRequest,
+  PipelineResponse,
+  SendRequest,
+} from "@azure/core-rest-pipeline";
+
+const APIM_KEY_HEADER_NAME = "Ocp-Apim-Subscription-Key";
+const APIM_REGION_HEADER_NAME = "Ocp-Apim-Subscription-Region";
+
+export class TranslatorCredential {
```

Comment:
Ideally, we should be able to make it without a class and just have the interface. The class is not providing any benefits here. I would recommend this to be

```ts
export interface TranslatorCredential {
   key: string,
   region: string
}
```

If we really need to have the class, we should have the interface above and the class implementing this interface. Also, the createClient factory function should take the interface as parameter instead of the class instance

Reply:
ok, done